components/eamxx/src/share/io/scream_scorpio_interface.cpp

```diff
@@ -1204,19 +1204,6 @@ void read_var (const std::string &filename, const std::string &varname, T* buf,
         " - time len: " + std::to_string(f.time_dim->length));
     err = PIOc_setframe(f.ncid,var.ncid,frame);
     check_scorpio_noerr (err,f.name,"variable",varname,"read_var","setframe");
-  } else if (time_index>=0) {
```

Comment:
This branch was at some point needed to allow IOP to read data with a time dimension that was not unlimited. However, since then, I have added `scorpio::pretend_dim_is_unlimited(..)`, which allows one to treat a normal dim as unlimited, and marking all vars with that dim as "time_dep". This ensures that the branch above is used for these variables. This is already used in the IOPDataManager structure to handle this case.